### PR TITLE
refactor: remove redundant task-algo checks

### PR DESCRIPTION
**Pull request by mblottiere**
## Description

With this patch, we only rely on algo and task inputs/outputs matching.
This means we can get rid of redundant checks:
- task / algo category matching
- parent task category check

## How has this been tested?

- relying on CI

## Checklist

- [x] [changelog](../CHANGELOG.md) was updated with notable changes
- [x] documentation was updated

### CODE REVIEW

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: docs/assets/computetask.md [hunk 1]
```diff
@@ -16,33 +16,26 @@ Those models will have different [permissions](./permissions.md), which will be
...
 
-**Note**:
+On task creation, inputs should match the expectations of the algorithm used:
 
-- the asterisk denotes an exclusive link, ie a *Test* task can only have **one** parent at most
-- parenthesis denotes optional dependencies
-- no parents may be a valid input
+- mandatory inputs should be defined
+- asset kind should match
+- permissions should allow processing
```
Comment (AurelienGasser) [inline docs/assets/computetask.md:35]: This item feels different from the others. Maybe move it to a standalone sentence?

File: docs/assets/computetask.md [hunk 2]
```diff
@@ -16,33 +16,26 @@ Those models will have different [permissions](./permissions.md), which will be
...
-| Test                  | 1*    | 0    | 1*        | 1*        | 1*      |
-| Aggregate             | n     | 0    | n         | n         | 0       |
-| Composite             | 0     | 0    | (1)       | 1, (2)    | 0       |
-| Predict               | 1*    | 0    | 1*        | 1*        | 0       |
-
-A Composite task may have:
-- no parents
-- an aggregate and a composite (head will come from the composite)
-- two composites (head will come from the **first parent task**)
+For legacy reasons, a task cannot receive more than one datamanager
```
Comment (AurelienGasser) [inline docs/assets/computetask.md:37]: I don't think it's legacy, it's more like the current (intentional) design. What do you think?

```suggestion
A task cannot have than one datamanager as an input.
```

**Review by AurelienGasser — APPROVED** [review]
👍

File: docs/assets/computetask.md [hunk 3]
```diff
@@ -16,33 +16,26 @@ Those models will have different [permissions](./permissions.md), which will be
...
+Upon successful execution, compute task should generate *outputs* which correspond to the outputs defined in their [Algo](./algo.md).
+Each compute task output has a separate set of [Permissions](../permissions.md).
+The permissions must be set for each output at task creation,
+with the exception of outputs of kind "performance" which are always public and for which setting specific permissions is not allowed.
 
 ## Compatibility
 
-Since a compute task will receive models from their parents,
-there are some restrictions on which parents are allowed for each task.
+Tasks receives explicit inputs, they can be either:
```
Comment (thibowk) [inline docs/assets/computetask.md:26]:
```suggestion
Tasks receive explicit inputs, they can be either:
```

**Pull request merged**